File: FoodOn_backend/src/main/java/com/foodon/foodon/recommend/dto/RecommendFoodResponse.java
package com.foodon.foodon.recommend.dto;

import com.foodon.foodon.food.dto.NutrientClaimInfo;
import com.foodon.foodon.food.dto.response.NutrientClaimResponse;
import com.foodon.foodon.recommend.domain.RecommendFood;

import java.math.BigDecimal;
import java.util.List;

public record RecommendFoodResponse(
        Long foodId,
        String foodName,
        BigDecimal kcal,
        String reason,
        List<NutrientClaimResponse> nutrientClaims
) {
    public static RecommendFoodResponse from(
            RecommendFood recommendFood,
            List<NutrientClaimInfo> nutrientClaims
    ) {
        return new RecommendFoodResponse(
                recommendFood.getFoodId(),
                recommendFood.getFoodName(),
                recommendFood.getKcalPerServing(),
                recommendFood.getReason(),
                nutrientClaims.stream()
                        .map(NutrientClaimInfo::type)
                        .map(NutrientClaimResponse::from)
                        .toList()
        );
    }
}

File: android-app/app/src/main/java/com/swallaby/foodon/data/calendar/remote/dto/RecommendFoodResponse.kt
package com.swallaby.foodon.data.calendar.remote.dto

import com.swallaby.foodon.domain.calendar.model.RecommendFood
import com.swallaby.foodon.domain.food.model.NutrientClaimInfo

data class RecommendFoodResponse(
    val foodId: Long = 0,
    val foodName: String = "",
    val kcal: Int = 0,
    val nutrientClaims: List<NutrientClaimInfo> = emptyList()
)

fun RecommendFoodResponse.toDomain(): RecommendFood {
    return RecommendFood(
        foodId = this.foodId,
        name = this.foodName,
        kcal = this.kcal,
        nutrientClaims = this.nutrientClaims
    )
}
